code/modules/food_and_drinks/machinery/stove.dm

```diff
@@ -115,6 +115,25 @@
 	. = ..()
 	LAZYREMOVE(added_ingredients, gone)
 
+/obj/item/reagent_containers/cup/soup_pot/proc/containeradd(obj/item/attacking_item, mob/user, params)
```

Comment:
Document this proc?

Reply:
I think it could use a better name though, perhaps `transfer_from_container_to_pot`. 

And the args - rename `attacking_item` to `transfer_from`, and delete `params` (you don't pass it anyways)